eng/common/pipelines/templates/steps/verify-links.yml

```diff
@@ -11,4 +11,10 @@ steps:
       workingDirectory: $(Build.SourcesDirectory)/${{ parameters.Directory }}
       filePath: eng/common/scripts/Verify-Links.ps1 
       arguments: >
-        -urls $(dir -r -i *.md) -rootUrl "file://$(Build.SourcesDirectory)/${{ parameters.Directory }}" -recursive:$false -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
+        -urls $(dir -r -i *.md) 
+        -rootUrl "file://$(Build.SourcesDirectory)/${{ parameters.Directory }}" 
+        -recursive $false 
+        -ignoreLinksFile ${{ parameters.IgnoreLinksFile }}
+        -branchReplaceRegex "($(System.PullRequest.sourceRepositoryUri)/blob/)master(/.*)"
+        -branchReplacementName $(system.pullRequest.sourceCommitId)
```

Comment:
One more validation I'd like you to perform is what happens when this template gets used in a scheduled or manually ran build that isn't associated with a pull request. I suspect these values will not be replaced and become literal strings `$(system.pullRequest.sourceCommitId)` which sill then fail in the PS script. Can you please verify the behavior?

Have a read over https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch#understand-variable-syntax to better understand the variable macro expansion problem.

Reply:
Tested manually trigger pipeline with $(System.PullRequest.sourceRepositoryUri) and $(system.pullRequest.sourceCommitId).
The pipeline throw us error when it is not a pull request.
Will use try/catch for the two variables.

Reply:
I don't think try catch is the answer here. I'm not sure if it will work but can you try accessing the variables via `variables['system.pullRequest.sourceCommitId']`? It isn't clear to me from the docs if you can only access them this way conditions or not so it will require some testing.

Reply:
It is not working. I am trying to add one more step of checking whether it is triggered by PR. Testing in PR here:
https://github.com/Azure/azure-sdk-for-android/pull/289